sktime/datatypes/_check.py

```diff
@@ -302,16 +304,16 @@ def mtype(
     mtypes_negative = dict()
 
     for m_plus_scitype in m_plus_scitypes:
-        valid, msg, _ = check_is_mtype(
+        valid = check_is_mtype(
             obj,
             mtype=m_plus_scitype[0],
             scitype=m_plus_scitype[1],
-            return_metadata=True,
+            return_metadata=False,
         )
         if valid:
             mtypes_positive += [m_plus_scitype[0]]
         else:
-            mtypes_negative[m_plus_scitype[0]] = msg
+            mtypes_negative[m_plus_scitype[0]] = ""
```

Comment:
I see what you're doing - very smart observation imo.

The "clean" way would be a refactor with deprecation.
One way to do it would be to introduce a `return_msg` argument everywhere, which can be True, False, or None (default).
If None, the current behaviour happens and a deprecation warning is raised. If `True`, returns the `msg` return, independently of `return_metadata`. If `False`, never returns the `msg` arg.
After a deprecation cycle, the default changes to `False`.

Reply:
it is nightmarish though since it would have to be one in *all the checker functions*.

I would definitely advise a separate PR for this with only that as the purpose, since PRs should be focused in file footprint, or in scope. It is not possible to focus this in terms of file footprint...

unless you come up with a better idea.

Reply:
Moved discussion to #4191